packages/ethers5/src/client.ts

```diff
@@ -168,8 +170,36 @@ export class Web3Modal extends Web3ModalScaffold {
         WalletConnectProvider.on('display_uri', (uri: string) => {
           onUri(uri)
         })
+        console.log('SIWEController', SIWEController, siweConfig)
+        console.log('WalletConnectProvider init opts', options)
+
+        if (siweConfig?.options?.enabled) {
+          const result = await WalletConnectProvider.authenticate({
+            nonce: await siweConfig.getNonce(),
+            methods: OPTIONAL_METHODS,
+            ...siweConfig.options.messageParams
+          })
+          if (result?.auths?.[0]) {
+            const chainId = parseInt(result.auths[0].p.chainId?.split(':')?.[1] || '', 10)
+            const address = result.auths[0]?.p.iss.split(':')[2] || ''
+            SIWEController.setSession({
+              address,
+              chainId
+            })
 
-        await WalletConnectProvider.connect()
+            // Kicks off verifyMessage and populates external states
+            const message = WalletConnectProvider.signer.client.formatAuthMessage({
+              request: result.auths[0].p,
```

Comment:
What does` result.auths[0]` represent ? Can we use a variable with an explicit name ?